compiler/ETA/CodeGen/Layout.hs

```diff
@@ -142,28 +143,52 @@ slowCall fun args = do
         ft = locFt fun
         code = loadLoc fun
 
-directCall :: Bool -> Code -> RepArity -> [StgArg] -> CodeGen ()
-directCall slow entryCode arity args = do
+directCall :: Bool -> CgLoc -> RepArity -> [StgArg] -> CodeGen ()
+directCall slow cgLoc@(LocStatic _ modClass clName) arity args = do
+  argFtCodes <- getRepFtCodes args
+  emit $ directStaticCall slow staticClosure modClass' arity argFtCodes
+  where staticClosure =
+             loadLoc cgLoc
+          <> pop closureType
```

Comment:
I originally had this as:

```haskell
gconv closureType (obj (innerClas modClass clName))
```

Casts are meant to be fast on the JVM. Faster than this field access? Not sure which it should be.